common/static/common/js/components/collections/paging_collection.js

```diff
@@ -25,8 +25,31 @@
                 // to the same object references for sortableFields and filterableFields.
                 this.sortableFields = {};
                 this.filterableFields = {};
+
+                this.paginator_core = {
+                    type: 'GET',
+                        dataType: 'json',
+                        url: function () { return this.url; }
+                };
+                var self = this;
+                this.paginator_ui = {
+                    firstPage: function () { return self.isZeroIndexed ? 0 : 1; },
+                    // Specifies the initial page during collection initialization
+                    currentPage: self.isZeroIndexed ? 0 : 1,
+                    perPage: function () { return self.perPage; }
+                };
+
+                this.currentPage = this.paginator_ui.currentPage;
+
+                this.server_api = {
+                    page: function () { return self.currentPage; },
+                    page_size: function () { return self.perPage; },
+                    text_search: function () { return self.searchString ? self.searchString : ''; },
+                    sort_order: function () { return self.sortField; }
+                };
             },
 
+            self: this,
```

Comment:
@andy-armstrong `this` is context-dependent. If I want to reference the outer scope's `this` instead of the innter-scope's `this`, I have to give it a name. The name `self` is typically used in our backbone views when such a case arises.